lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -11,84 +11,1938 @@
+        _.defineProperty(collator_supportedLocalesOf, "name", {
+            value: "supportedLocalesOf",
+            writable: false,
+            enumerable: false,
+            configurable: true,
+        });
+        _.defineProperty(Collator, "supportedLocalesOf", {
+            value: collator_supportedLocalesOf,
+            writable: true,
+            enumerable: false,
+            configurable: true,
+        });
+
+        _.defineProperty(Collator, "prototype", {
+            value: new Collator(),
+            writable: false,
+            enumerable: false,
+            configurable: false
+        });
+        setPrototype(Collator.prototype, Object.prototype);
+
+        _.defineProperty(Collator.prototype, "constructor", {
+            value: Collator,
+            writable: true,
+            enumerable: false,
+            configurable: true
+        });
+        _.defineProperty(Collator.prototype, "resolvedOptions", {
+            value: function resolvedOptions() {
+                if (typeof this !== "object") {
+                    platform.raiseNeedObjectOfType("Collator.prototype.resolvedOptions", "Collator");
+                }
+                const hiddenObject = platform.getHiddenObject(this);
+                if (hiddenObject === undefined || !hiddenObject.initializedCollator) {
+                    platform.raiseNeedObjectOfType("Collator.prototype.resolvedOptions", "Collator");
+                }
+
+                const options = [
+                    "locale",
+                    "usage",
+                    "sensitivity",
+                    "ignorePunctuation",
+                    "collation",
+                    "numeric",
+                    "caseFirst",
+                ];
+
+                return createResolvedOptions(options, hiddenObject);
+            },
+            writable: true,
+            enumerable: false,
+            configurable: true
+        });
+
+        // test262's test\intl402\Collator\prototype\compare\name.js checks the name of the descriptor's getter function
+        const getCompare = function () {
+            if (typeof this !== "object") {
+                platform.raiseNeedObjectOfType("Collator.prototype.compare", "Collator");
+            }
+
+            const hiddenObject = platform.getHiddenObject(this);
+            if (hiddenObject === undefined || !hiddenObject.initializedCollator) {
+                platform.raiseNeedObjectOfType("Collator.prototype.compare", "Collator");
+            }
+
+            return hiddenObject.boundCompare;
+        };
+        _.defineProperty(getCompare, "name", {
+            value: "get compare",
+            writable: false,
+            enumerable: false,
+            configurable: true,
+        });
+        _.defineProperty(Collator.prototype, "compare", {
+            get: tagPublicFunction("get compare", getCompare),
```

Comment:
Is tagging a function still required if we have redefined its name?

Reply:
I think so. I believe it does more than just change the name. Things like marking it as a built-in library function as opposed to a user function.